### PR TITLE
PR to use ViewBinding for Android ArcGIS Samples

### PR DESCRIPTION
**Pull request by shubham7109**
PR to update Android ArcGIS samples to use ViewBinding development methodology since Kotlin Android Extensions will be deprecated soon. 

Some samples have errors during runtime which I have addressed. in the comments. They require further testing, rest have been tested with the ViewBinding changes. 

- `animate-images-with-image-overlay`
Sample viewer crashes when downloading offline data

- `configure-subnetwork-trace`
404 Page not found URL is used

- `display-utility-associations`
Runtime error

- `edit-with-branch-versioning`
404 Page not found URL is used

- `offline-geocode`
Data does not download in the sample viewer

- `perform-valve-isolation-trace`
404 Page found not URL is used

- `show-labels-on-layer-in-3d`
Sample viewer does not show up on search

- `trace-utility-network`
404 Page not found URL is used

- `viewshed-geoprocessing`
Geo-processing result fails

This PR is riddled with changes, so I understand if it will take time to review them. I will make sure to keep the PR up-to date and work on your comments/suggestions. Thanks!

### CODE REVIEW

File: kotlin/apply-mosaic-rule-to-rasters/src/main/java/com/esri/arcgisruntime/sample/applymosaicruletorasters/MainActivity.kt [hunk 1]
```diff
@@ -22,23 +22,37 @@ import android.util.Log
...
+    activityMainBinding.mapView
+  }
+
+  private val spinner: Spinner by lazy {
+    activityMainBinding.spinner
+  }
+
   // create an image service raster from a url for an image service
   private val imageServiceRaster: ImageServiceRaster by lazy {
     ImageServiceRaster("https://sampleserver7.arcgisonline.com/arcgis/rest/services/amberg_germany/ImageServer").apply {
```
Comment (shubham7109) [inline kotlin/apply-mosaic-rule-to-rasters/src/main/java/com/esri/arcgisruntime/sample/applymosaicruletorasters/MainActivity.kt:58]: This line returns a 404 error. Hence sample loads but fails to retrieve the `ImageServiceRaster`

File: kotlin/configure-subnetwork-trace/src/main/java/com/esri/arcgisruntime/sample/configuresubnetworktrace/MainActivity.kt [hunk 2]
```diff
@@ -29,13 +35,47 @@ import com.esri.arcgisruntime.utilitynetworks.UtilityTraceParameters
...
+  private val valueBooleanButton: ToggleButton by lazy {
+    activityMainBinding.valueBooleanButton
+  }
+
+  private val valuesEditText: EditText by lazy {
+    activityMainBinding.valuesEditText
+  }
+
   private val utilityNetwork by lazy {
     UtilityNetwork(getString(R.string.naperville_electric_url))
```
Comment (shubham7109) [inline kotlin/configure-subnetwork-trace/src/main/java/com/esri/arcgisruntime/sample/configuresubnetworktrace/MainActivity.kt:80]: The `naperville_electric_url` URL returns a 404 error. Sample crashes.

File: kotlin/edit-with-branch-versioning/src/main/java/com/esri/arcgisruntime/sample/editwithbranchversioning/MainActivity.kt [hunk 3]
```diff
@@ -40,336 +38,373 @@ import com.esri.arcgisruntime.mapping.ArcGISMap
...
-            val screenPoint = android.graphics.Point(e.x.toInt(), e.y.toInt())
+    private val TAG = MainActivity::class.java.simpleName
 
-            // if selected feature is not null, edit the feature location
-            selectedFeature?.let {
-              editFeatureLocation(it, screenPoint)
-              return true
-            }
+    private val serviceGeodatabase: ServiceGeodatabase by lazy {
+        ServiceGeodatabase("https://sampleserver7.arcgisonline.com/arcgis/rest/services/DamageAssessment/FeatureServer")
```
Comment (shubham7109) [inline kotlin/edit-with-branch-versioning/src/main/java/com/esri/arcgisruntime/sample/editwithbranchversioning/MainActivity.kt:51]: Link points to a 404 error page.

File: kotlin/perform-valve-isolation-trace/src/main/java/com/esri/arcgisruntime/sample/performvalveisolationtrace/MainActivity.kt [hunk 4]
```diff
@@ -53,276 +55,313 @@ import com.esri.arcgisruntime.utilitynetworks.UtilityTraceConfiguration
...
-          }
+
+    /**
+     * Create and load a utility network from the string resource url and initialize a starting point
+     * from it.
+     */
+    private fun createUtilityNetwork(
+    ) {
+        // create a utility network from the url and load it
+        utilityNetwork = UtilityNetwork(getString(R.string.utility_network_url), mapView.map)
```
Comment (shubham7109) [inline kotlin/perform-valve-isolation-trace/src/main/java/com/esri/arcgisruntime/sample/performvalveisolationtrace/MainActivity.kt:170]: Link points to a 404 error page

**Comment by TADraeseke** [comment]
Still struggling with getting the sample viewer to build from this branch. I've written logic to handle correct renaming of "ActivityMainBinding" classes at least: https://devtopia.esri.com/runtime/java/tree/trev8939/databindingInViewer

**Review by TADraeseke — APPROVED** [review]
Great work @shubham7109! I've been through every one of these samples in the sample viewer and they all work brilliantly. Well done!

The running list of Kotlin samples not yet on the new pattern I note as:
- Browse OGC API feature service
- Display OGC API Collection
- Display device location with nmea data sources
- Setup location driven geotriggers
- Show location history
- Query with CQL filter
- ArcGIS vector tiled layer (custom style)
- Create symbol styles from web styles
- Show labels on layer in 3D

but let's take these as a separate issue and separate PR!

Nice job!

**Comment by shubham7109** [comment]
Sounds good, thank you!

**Comment by shubham7109** [comment]
I have logged an issue to address the remaining samples! https://devtopia.esri.com/runtime/common-samples/issues/3150

**Pull request merged**